PacketDotNet/IPv6ExtensionHeader.cs

```diff
@@ -0,0 +1,70 @@
+﻿using PacketDotNet.MiscUtil.Conversion;
+using PacketDotNet.Utils;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+/*
+This file is part of PacketDotNet
+
+PacketDotNet is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+PacketDotNet is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with PacketDotNet.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/*
+ * Copyright 2018 Steven Haufe<haufes@hotmail.com>
+ */
+
+namespace PacketDotNet
+{
+    [Serializable]
+    internal class IPv6ExtensionHeader
+    {
+        static public List<IPProtocolType> extensionHeaderTypes = new List<IPProtocolType>{ IPProtocolType.HOPOPTS,
+            IPProtocolType.DSTOPTS,
+            IPProtocolType.ROUTING,
+            IPProtocolType.FRAGMENT,
+            IPProtocolType.AH,
+            IPProtocolType.ENCAP,
+            IPProtocolType.DSTOPTS,
+            IPProtocolType.MOBILITY,
+            IPProtocolType.HOSTIDENTITY,
+            IPProtocolType.SHIM6 };
+
+
+        protected ByteArraySegment Header;
+
+
+        public IPv6ExtensionHeader(ByteArraySegment bas)
```

Comment:
I think it would be nice to have at least a NextHeader property defined here to the underlying data. This would also open up the possibility to parse the actual data, as as-is it's just a block of data with no actual meaning. What I'm proposing is a similar structure as the TcpOptions, see https://github.com/chmorgan/packetnet/blob/master/PacketDotNet/Tcp/Option.cs. Then this class would need to become abstract and public.

Reply:
I'll add the NextHeader in. Manipulating the headers is tricky because they way they are linked, I'm just trying to fix the parsing of the packet with this commit.